ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -246,28 +239,58 @@ private[state] object Conversions {
   /** Encodes a [[BlindingInfo]] into protobuf (i.e., [[DamlTransactionBlindingInfo]]).
     * It is consensus-safe because it does so deterministically.
     */
-  def encodeBlindingInfo(blindingInfo: BlindingInfo): DamlTransactionBlindingInfo =
+  def encodeBlindingInfo(
+      blindingInfo: BlindingInfo,
+      divulgedContracts: Map[ContractId, TransactionOuterClass.ContractInstance],
+  ): DamlTransactionBlindingInfo =
     DamlTransactionBlindingInfo.newBuilder
       .addAllDisclosures(encodeDisclosure(blindingInfo.disclosure).asJava)
-      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence).asJava)
+      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence, divulgedContracts).asJava)
       .build
 
-  def decodeBlindingInfo(blindingInfo: DamlTransactionBlindingInfo): BlindingInfo =
-    BlindingInfo(
-      disclosure = blindingInfo.getDisclosuresList.asScala.map { disclosureEntry =>
+  def decodeBlindingInfo(
+      damlTransactionBlindingInfo: DamlTransactionBlindingInfo
+  ): (BlindingInfo, Map[ContractId, Option[Value.ContractInst[VersionedValue[ContractId]]]]) = {
```

Comment:
Isn't it the case that, for a given decoding, either all the instances are there or none of them are?

Reply:
Yes that is true. Initially I wanted to separate the concerns of extraction/validation and have the ability to log them if any are missing. On a second thought, I see that it might not bring much value. Adapted to your recommendation: the refactored utility function `extractDivulgedContracts` which now returns this map, now returns an `Option[Map[..]]`